### PR TITLE
Add script to update hub code from foundry

### PR DESCRIPTION
**Pull request by dakinggg**
Adds a script to open PRs against huggingface hub models to keep in sync with foundry code.

TODO:
- [x] add a "test uploaded model" arg
- [x] run and open the prs

Opened some prs with this script:
https://huggingface.co/mosaicml/mpt-7b/discussions/63
https://huggingface.co/mosaicml/mpt-7b-instruct/discussions/51
https://huggingface.co/mosaicml/mpt-7b-chat/discussions/29
https://huggingface.co/mosaicml/mpt-7b-storywriter/discussions/47

### CODE REVIEW

File: scripts/misc/update_hub_code.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+            _ = pr_tokenizer.batch_decode(generation)
+
+            print(f'Opening PR against {repo}')
+            result = api.upload_folder(
+                folder_path=original_save_dir,
+                repo_id=repo,
+                use_auth_token=True,
+                repo_type='model',
+                allow_patterns=['*.py'],
+                commit_message='LLM-foundry update',
```
Comment (abhi-mosaic) [inline scripts/misc/update_hub_code.py:56]: Could we add a datetime here?

**Review by abhi-mosaic — APPROVED** [review]
This LGTM, I am kinda confused by the usage of `torch.nn.Embedding` rather than `nn.Embedding` in the HF Hub PR https://huggingface.co/mosaicml/mpt-7b/discussions/61/files but it's not breaking anything. If there's an easy way to fix that in this PR and push it up too, that would be cool. But not gating.

**Comment by dakinggg** [comment]
our main code uses `torch.nn.Embedding`, at some point someone updated the hub code to be `nn.Embedding`, this is just bringing them back in sync. doesn't really matter which one so ill just leave it as `torch.nn.Embedding`

File: scripts/misc/update_hub_code.py (hunk 1 above)
Comment (dakinggg) [reply scripts/misc/update_hub_code.py]: added

**Pull request merged**